wagtail/admin/views/collections.py

```diff
@@ -57,26 +64,59 @@ class Edit(EditView):
     context_object_name = 'collection'
     header_icon = 'folder-open-1'
 
+    def _user_may_move_collection(self, user, instance):
+        """
+        Is this instance used for assigning GroupCollectionPermissions to the user?
+        If so, this user may not move the collection to a new part of the tree
+        """
+        if user.is_active and user.is_superuser:
+            return True
+        else:
+            permissions = self.permission_policy._get_permission_objects_for_actions(['add', 'edit', 'delete'])
+            return not GroupCollectionPermission.objects.filter(
+                group__user=user,
+                permission__in=permissions,
+                collection=instance,
+            ).exists()
+
+    def get_queryset(self):
+        return self.permission_policy.instances_user_has_permission_for(
+            self.request.user, 'change'
+        ).exclude(depth=1)
+
+    def get_form(self, form_class=None):
+        form = super().get_form(form_class)
+        user = self.request.user
+        # if user does not have add permission anywhere, they can't move a collection
+        if not self.permission_policy.user_has_permission(user, 'add'):
+            form.fields.pop('parent')
+        # If this instance is a collection used to assign permissions for this user,
+        # do not let the user move this collection.
+        elif not self._user_may_move_collection(user, form.instance):
+            form.fields.pop('parent')
+        else:
+            # Filter collections offered in parent field by current user's add permissions
+            collections = self.permission_policy.instances_user_has_permission_for(user, 'add')
+            form.fields['parent'].queryset = collections
+            form.fields['parent'].disabled_queryset = form.instance.get_descendants(inclusive=True)
```

Comment:
Is this attribute still used?

Reply:
Yes.

Reply:
Ah, I see. `disabled_queryset` is property setter that affects the widget choices!